packages/core/src/new-api/internal/new-execution/type-guards/execution-state.ts

```diff
@@ -0,0 +1,82 @@
+import {
```

Comment:
As mentioned in the previous file, we shouldn't need this. Let's try not to add these functions. They add a ton of code and potentially hide bugs. Now everything has a `type`.